trade-finance-manager-api/src/v1/controllers/user/user.routes.test.js

```diff
@@ -0,0 +1,114 @@
+        const expectedErrorResponse = { message: 'Error validating payload', status: 400 };
+        await upsertTfmUserFromEntraIdUser(req, res, next);
+
+        expect(res._getData()).toEqual(expectedErrorResponse);
+        expect(res._getStatusCode()).toEqual(HttpStatusCode.BadRequest);
+      });
+    });
+
+    describe('when the request body is valid', () => {
+      let validRequest;
+
+      beforeEach(() => {
+        validRequest = aValidEntraIdUser();
+      });
+
+      it('calls the user controller', async () => {
+        const { req, res, next } = getHttpMocks(validRequest);
+
+        await upsertTfmUserFromEntraIdUser(req, res, next);
+
+        expect(jest.mocked(userController.upsertTfmUserFromEntraIdUser)).toHaveBeenCalledTimes(1);
+      });
+
+      describe('when the upsert is unsuccessful', () => {
+        describe('when the error is an api error', () => {
+          const apiErrorDetails = {
+            status: 500,
+            message: 'an api error message',
+            cause: 'a cause that should not be exposed',
+            code: 'a code',
+          };
+
+          beforeEach(() => {
+            jest.mocked(userController.upsertTfmUserFromEntraIdUser).mockRejectedValueOnce(new ApiError(apiErrorDetails));
+          });
+
+          it('returns the api error with expected formatting', async () => {
+            const { req, res, next } = getHttpMocks(validRequest);
+            const { cause, ...expectedErrorDetails } = apiErrorDetails;
+            await upsertTfmUserFromEntraIdUser(req, res, next);
+
+            expect(res._getData()).toEqual(expectedErrorDetails);
+            expect(res._getStatusCode()).toEqual(expectedErrorDetails.status);
+          });
+        });
+
+        describe('when the error is not an api error', () => {
+          const errorToThrow = new Error('An Error');
+          beforeEach(() => {
+            jest.mocked(userController.upsertTfmUserFromEntraIdUser).mockRejectedValueOnce(errorToThrow);
+          });
+
+          it('passes the error to further middleware', async () => {
+            const { req, res, next } = getHttpMocks(validRequest);
+
+            await upsertTfmUserFromEntraIdUser(req, res, next);
+
+            expect(next).toHaveBeenCalledWith(errorToThrow);
+          });
+        });
+      });
+
+      describe('when the upsert is successful', () => {
+        const aSuccessfulResponseFromController = 'aSuccessfulResponse';
+
+        beforeEach(() => {
+          jest.mocked(userController.upsertTfmUserFromEntraIdUser).mockResolvedValueOnce(aSuccessfulResponseFromController);
+        });
+
+        it('returns the result of the user controller', async () => {
+          const { req, res, next } = getHttpMocks(validRequest);
+
+          await upsertTfmUserFromEntraIdUser(req, res, next);
+
+          expect(res._getData()).toEqual(aSuccessfulResponseFromController);
+        });
+      });
+    });
+
+    function getHttpMocks(body) {
```

Comment:
Not a blocker for this PR, it looks like repeated or very similar approaches for this has been introduced in 5x places, if we search for `function getHttpMocks`, should be DRY